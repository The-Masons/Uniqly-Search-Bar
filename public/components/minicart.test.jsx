import React from 'react';
import MiniCart from './minicart.jsx';

describe('MiniCart', () => {
  test('should correctly render an empty cart', () => {
    const miniCart = shallow(<MiniCart />);
    const cartView = miniCart.find('.minicart-view');

    expect(cartView.hasClass('hidden')).toEqual(true);
    expect(cartView.hasClass('empty')).toEqual(true);
    expect(cartView.text()).toEqual('YOUR BAG IS EMPTY');
  });

  test('should correctly render a cart with items', () => {
    const miniCart = mount(<MiniCart
      cart={{
        'Fake 1': {
          name: 'Dog Shirt',
          color: 'Dog Color',
          quantity: 2,
          size: 'Small Dog Size',
          price: 9999,
          imgUrl: 'http://placecorgi.com/250'
        },
        'Fake 2': {
          name: 'Cat Shirt',
          color: 'Cat Color',
          quantity: 1,
          size: 'Cat Size',
          price: 5099,
          imgUrl: 'http://placecorgi.com/250'
        },
        'Fake 3': {
          name: 'Some Pants',
          color: 'Pants Color',
          quantity: 1,
          size: 'Human Size',
          price: 1599,
          imgUrl: 'http://placecorgi.com/250'
        },
      }}
      cartSize={4}
      cartOrder={['Fake 1', 'Fake 3', 'Fake 2']}
      />);

    expect(miniCart).toMatchSnapshot();
  });

  test('should display the cart when moused over', () => {
    const miniCart = mount(<MiniCart />);
    const cartIcon = miniCart.find('.minicart-icon');

    cartIcon.simulate('mouseEnter');
    expect(miniCart.state('viewClass')).toEqual('minicart-view empty');
  });

  test('should hide the cart when no longer moused over', () => {
    const miniCart = mount(<MiniCart />);
    const cartIcon = miniCart.find('.minicart-icon');
    jest.useFakeTimers();

    cartIcon.simulate('mouseEnter');
    cartIcon.simulate('mouseLeave');

    jest.advanceTimersByTime(5000);
    expect(miniCart.state('viewClass')).toEqual('minicart-view empty hidden');
  });
});
