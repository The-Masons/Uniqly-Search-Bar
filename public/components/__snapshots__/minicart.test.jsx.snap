// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCart should correctly render a cart with items 1`] = `
<MiniCart
  cart={
    Object {
      "Fake 1": Object {
        "color": "Dog Color",
        "imgUrl": "http://placecorgi.com/250",
        "name": "Dog Shirt",
        "price": 9999,
        "quantity": 2,
        "size": "Small Dog Size",
      },
      "Fake 2": Object {
        "color": "Cat Color",
        "imgUrl": "http://placecorgi.com/250",
        "name": "Cat Shirt",
        "price": 5099,
        "quantity": 1,
        "size": "Cat Size",
      },
      "Fake 3": Object {
        "color": "Pants Color",
        "imgUrl": "http://placecorgi.com/250",
        "name": "Some Pants",
        "price": 1599,
        "quantity": 1,
        "size": "Human Size",
      },
    }
  }
  cartOrder={
    Array [
      "Fake 1",
      "Fake 3",
      "Fake 2",
    ]
  }
  cartSize={4}
>
  <div
    className="minicart"
  >
    <div
      className="minicart-icon"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <span>
        4
      </span>
    </div>
    <div
      className="minicart-view hidden"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="minicart-user-cart"
        key="userCart"
      >
        <div
          className="cart-item"
          key="cartItem0"
        >
          <img
            className="cart-item-img"
            src="http://placecorgi.com/250"
          />
          <div
            className="cart-item-info"
          >
            <span
              className="cart-item-info name"
            >
              Dog Shirt
            </span>
            <span
              className="cart-item-info qty"
            >
              2
            </span>
            <span
              className="cart-item-info color"
            >
              Dog Color
            </span>
            <span
              className="cart-item-info size"
            >
              Small Dog Size
            </span>
          </div>
          <span
            className="cart-item-price"
          >
            $
            99.99
          </span>
        </div>
        <div
          className="cart-item"
          key="cartItem1"
        >
          <img
            className="cart-item-img"
            src="http://placecorgi.com/250"
          />
          <div
            className="cart-item-info"
          >
            <span
              className="cart-item-info name"
            >
              Some Pants
            </span>
            <span
              className="cart-item-info qty"
            >
              1
            </span>
            <span
              className="cart-item-info color"
            >
              Pants Color
            </span>
            <span
              className="cart-item-info size"
            >
              Human Size
            </span>
          </div>
          <span
            className="cart-item-price"
          >
            $
            15.99
          </span>
        </div>
        <div
          className="cart-item"
          key="cartItem2"
        >
          <img
            className="cart-item-img"
            src="http://placecorgi.com/250"
          />
          <div
            className="cart-item-info"
          >
            <span
              className="cart-item-info name"
            >
              Cat Shirt
            </span>
            <span
              className="cart-item-info qty"
            >
              1
            </span>
            <span
              className="cart-item-info color"
            >
              Cat Color
            </span>
            <span
              className="cart-item-info size"
            >
              Cat Size
            </span>
          </div>
          <span
            className="cart-item-price"
          >
            $
            50.99
          </span>
        </div>
      </div>
      <div
        className="minicart-total"
        key="cartTotal"
      >
        <span
          className="total-item-count"
        >
          TOTAL (
          4
           ITEMS)
        </span>
        <span
          className="total-subtotal"
        >
          $
          266.96
        </span>
      </div>
      <div
        className="minicart-cart-controls"
        key="cartControls"
      >
        <button
          className="minicart-view-cart-btn"
        >
          VIEW BAG
        </button>
        <button
          className="minicart-checkout-btn"
        >
          CHECKOUT
        </button>
      </div>
      <span
        className="minicart-close-btn"
        key="closeBag"
      >
        CLOSE BAG
      </span>
    </div>
  </div>
</MiniCart>
`;
